package/samplers/pfns4bo/sampler.py

```diff
@@ -0,0 +1,274 @@
+            "num_features": 18,
+            "hyperparameters": {
+                "lengthscale_concentration": 1.2106559584074301,
+                "lengthscale_rate": 1.5212245992840594,
+                "outputscale_concentration": 0.8452312502679863,
+                "outputscale_rate": 0.3993553245745406,
+                "add_linear_kernel": False,
+                "power_normalization": False,
+                "hebo_warping": False,
+                "unused_feature_likelihood": 0.3,
+                "observation_noise": True,
+            },
+        },
+        "epochs": 50,
+        "lr": 0.0001,
+        "seq_len": 60,
+        "single_eval_pos_gen": utils.get_uniform_single_eval_pos_sampler(
+            50, min_len=1
+        ),  # <function utils.get_uniform_single_eval_pos_sampler.<locals>.<lambda>()>,
+        "aggregate_k_gradients": 2,
+        "nhid": 1024,
+        "steps_per_epoch": 1024,
+        "weight_decay": 0.0,
+        "train_mixed_precision": False,
+        "efficient_eval_masking": True,
+        "nlayers": 12,
+    }
+
+    bs = 128
+    all_targets = []
+    for num_hps in [
+        2,
+        8,
+        12,
+    ]:  # a few different samples in case the number of features makes a difference in y dist
+        b = config["priordataloader_class_or_get_batch"].get_batch_method(
+            bs,
+            1000,
+            num_hps,
+            epoch=0,
+            device=device,
+            hyperparameters={
+                **config["extra_prior_kwargs_dict"]["hyperparameters"],
+                "num_hyperparameter_samples_per_batch": -1,
+            },
+        )
+        all_targets.append(b.target_y.flatten())
+    ys = torch.cat(all_targets, 0).cpu()
+
+    config["criterion"] = bar_distribution.FullSupportBarDistribution(
+        bar_distribution.get_bucket_limits(1000, ys=ys)
+    )
+
+    return config
+
+
+class PFNs4BOSampler(BaseSampler):
+    def __init__(
+        self,
+        *,
+        prior: str | torch.nn.Module = "hebo",
+        model_path: str | None = None,
+        seed: int | None = None,
+        independent_sampler: BaseSampler | None = None,
+        n_startup_trials: int = 10,
+        num_grad_steps: int = 15_000,
+        num_candidates: int = 100,
+        pre_sample_size: int = 100_000,
+    ) -> None:
+        self._num_grad_steps = num_grad_steps
+        self._num_candidates = num_candidates
+        self._pre_sample_size = pre_sample_size
+
+        self._rng = LazyRandomState(seed)
+        self._independent_sampler = independent_sampler or RandomSampler(seed=seed)
+        self._intersection_search_space = IntersectionSearchSpace()
+        self._n_startup_trials = n_startup_trials
+
+        self._device = utils.default_device
+
+        if isinstance(prior, torch.nn.Module):
+            trained_model = prior
+        elif prior == "vanilla gp":
+            _, _, trained_model, _ = train(**get_vanilla_gp_config(self._device))
```

Comment:
I am not sure what your application is exactly. But training a PFN like the pre-trained ones takes around 1 day *with a GPU*. Thus, I don't think it is realistic HPO users want to do this, but they probably just want to use a pre-trained model. Either the ones I provided, or if you like to train better ones, then these.

Reply:
Sounds  very good. I completely agree with you. Since we should provide the pre-trained model for users, I will change the default value of the prior to use that provided in [here](https://github.com/automl/PFNs/blob/main/models_diff/prior_diff_real_checkpoint_n_0_epoch_42.cpkt).

Reply:
No these: https://github.com/automl/PFNs4BO/tree/main/pfns4bo/final_models